### DISCUSSION TITLE
Look into copilot alternatives like gpt-code-clippy and Captain Stack

### DISCUSSION THREAD

**Discussion by luisdavim** [neoclide/coc.nvim#3264] [General]
There are some open source alternatives to Github's copilot that provide VSCode extensions, would it be possible to convert some to coc extensions?
* https://github.com/ncoop57/gpt-code-clippy - similar to copilot
* https://github.com/hieunc229/copilot-clone - uses stackoverflow
* https://github.com/badboysm890/clara-copilot

**Comment by luisdavim** [comment]
Hi, is there a doc or a tool for converting vscode extensions into coc extensions?

**Comment by fannheyward** [comment]
I tested the https://github.com/ncoop57/code-clippy-vscode in VSCode, it's far away to use in real production code, does this work for you?

**Comment by luisdavim** [comment]
This one also seems interesting: https://github.com/tom-doerr/vim_codex

**Comment by sisrfeng** [comment]
https://github.com/github/copilot.vim

**Comment by sisrfeng** [comment]
I am using copilot, it can generate some codes and comments according to the context.
Sometimes it is useful, sometimes maybe annoying with unwanted suggestions.

Maybe the market share of Tabnine will be replaced by coplilot?
However, I can not find the feature of creating my own snippet in copilot, but success in Tabnine (with coc).

**Comment by luisdavim** [comment]
https://github.com/THUDM/CodeGeeX/issues/34#issuecomment-1360202055

**Comment by luisdavim** [comment]
There are a lot of alternatives now, including some that can run locally, some even act as a LSP and that makes integrating with COC much easier, for example https://tabby.tabbyml.com/blog/2024/02/05/create-tabby-extension-with-language-server-protocol/
